packages/tracing/src/hubextensions.ts

```diff
@@ -19,25 +20,68 @@ function traceHeaders(this: Hub): { [key: string]: string } {
 }
 
 /**
- * Use RNG to generate sampling decision, which all child spans inherit.
+ * Use sample rate (given in options as either a constant or ) along with a random number generator to make a sampling
+ * decision, which all child spans and child transactions inherit.
+ *
+ * Sample rate is set in SDK config, either as a constant (`tracesSampleRate`) or a function to compute the rate
+ * (`tracesSampler`).
+ *
+ * Called every time a transaction is created. Only transactions which emerge with sampled === true will be sent to
+ * Sentry.
+ *
+ * Mutates the given Transaction object and then returns the mutated object.
  */
 function sample<T extends Transaction>(hub: Hub, transaction: T): T {
   const client = hub.getClient();
+  const options = (client && client.getOptions()) || {};
```

Comment:
This could be written as:

```js
const client = hub.getClient();

// I assume if there's no client, there's no point in doing the rest of operations, as they all depend on the existence of options anyway
if (!client) {
  return transaction;
}

// `getOptions` always return at least an empty object. It's impossible to initialize the client otherwise.
const options = client.getOptions();

Reply:
True. It replaces 1 line with 5 lines, though...
![image](https://user-images.githubusercontent.com/14812505/90264014-4e6a1800-de05-11ea-9a3d-180d6cd7f0af.png)

There is something nice about acknowledging that not having a client matters, though. What about this?
![image](https://user-images.githubusercontent.com/14812505/90264282-b4569f80-de05-11ea-9a94-64c46457fef4.png)